### DISCUSSION TITLE
Trame dynamic table example with `vuetify3` doesn't show the table header

### DISCUSSION THREAD

**Discussion by kerim371** [Kitware/trame#581] [Q&A] [answered]
Hi,

I just modified Trame dyn table [example](https://github.com/Kitware/trame/blob/master/examples/00_howdoi/table_dyna.py) to use vuetify3 instead:
```python
from trame.app import get_server
from trame.ui.vuetify3 import SinglePageLayout
from trame.widgets import vuetify3 as vuetify
import json

import pandas as pd

# server = get_server(client_type="vue2")
server = get_server(client_type="vue3")
state = server.state
``` 
and I got table header disappeared like in the picture:
![image](https://github.com/user-attachments/assets/3517d43e-2e12-41af-8747-86fe6a057fa3)

**Comment by jourdain** [comment]
API between veutify2 and 3 are different. Adjust the keys to match the new API.

**Comment by kerim371** [comment]
Sorry didn't get that.

I just tried to add `"item_key": "Dessert"`:
```python
table = {
    "headers": ("headers", headers),
    "items": ("rows", rows),
    "search": ("query", ""),
    "classes": "elevation-1 ma-4",
    "multi_sort": True,
    "dense": True,
    "items_per_page": 5,
    "item_key": "Dessert"
}
```
but didn't work.


I also look in [vuetify table doc](https://trame.readthedocs.io/en/latest/trame.widgets.vuetify3.html#trame.widgets.vuetify3.VDataTable) and it looks a little bit awkward. Maybe I look in wrong resources?
![image](https://github.com/user-attachments/assets/430b9405-f5c6-4950-bd6b-46e73a1ebfcc)

**Reply by jourdain** [reply, marked as answer]
You can find a vue3 example with table [here](https://github.com/Kitware/trame-vuetify/blob/master/examples/vue3/datatable.py)

**Reply by jourdain** [reply]
The Python API/doc get generated from the JS doc. I usually directly look at the JS doc https://vuetifyjs.com/en/api/v-data-table/

**Reply by kerim371** [reply]
Thank you for the materials.

How do you think, if I just started using Trame user should I use Vuetify2 or Vuetify3? I guess Trame with Vuetify2 is more stable and a little bit easier to learn. But maybe there are some Vuetify3 advantages?

**Reply by jourdain** [reply]
vue3 is the way forward. Vuetify managed to catchup most of the feature it had in v2. So starting with vue3 should be fine now.

**Reply by kerim371** [reply]
Good to know! thank you!
